jobfunnel/jobfunnel.py

```diff
@@ -73,6 +75,15 @@ def __init__(self, args):
         if args['delay_config'] is not None:
             self.delay_config = args['delay_config']
 
+        # set session with (potential proxy)
+        self.s = Session()
+
+        # set proxy if given
+        if args['proxy'] is not None:
```

Comment:
I'm curious how calling `args['proxy']` does not error out with `KeyError: 'proxy'` :confused:.
Usually I would imagine this being:
```Python
if 'proxy' in args and args['proxy'] is not None:
    # do something
```
Anyone care to enlighten me?

Reply:
Well since these come from argparse, it will pass that flag as false if its not passed and true if it is - so always there. 

That said what you suggest is safer.

Reply:
By default, trying to access a dictionary key that does not exist results in a KeyError. However, in the `argparse` I use default_yaml.get('proxy', None) instead of default_yaml['proxy']. The former will return a `None` when the `proxy` key does not exist. 

Doing this, we are sure that `args['proxy']` does not throw a key value because it has value `None` if it is not set.

Reply:
I see.  So you don't need to implement my change in that case.  I didn't know that using `get` you could default to `None` if no config is found in the .yml.  Good stuff 👍

Reply:
The `collections.defaultdict` could also be useful for setting the `config` dictionary. The defaultdict allows to set a default value in case a key is missing. If you have a defaultdict `dic` without a key 'missing', then `dic['missing']` will return your specified default value, e.g. `None`.

I did not do this for now but it could potentially be of use in the future.